### PR TITLE
python27Packages.gtkme: init at version 1.5.1

### PR DESCRIPTION
**Pull request by tanneberger**
Adding Python2 Package gtkme

###### Motivation for this change
Some inkscape plugins need this package. 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
  - [x] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python2-modules/gtkme/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{ pkgs
+, lib
+, pkg-config
+, buildPythonPackage
+, buildPackages
+, fetchPypi
+, ...
```
Comment (veprbl) [inline pkgs/development/python2-modules/gtkme/default.nix:7]:
```suggestion
```

File: pkgs/development/python2-modules/gtkme/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  pname = "gtkme";
+  version = "1.5.3";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-NIUgnbfcHjbPfsH3CF2Bywo8owrdsi1wqDoMxOa+2U4=";
+  };
+
+  nativeBuildInputs = [ pkg-config pkgs.gobject-introspection ];
+  buildInputs = with pkgs.python27Packages; [ pygobject3 ] ++ [ pkgs.gtk3 ];
```
Comment (veprbl) [inline pkgs/development/python2-modules/gtkme/default.nix:20]:
```suggestion
  buildInputs = [ pygobject3 pkgs.gtk3 ];
```

File: pkgs/development/python2-modules/gtkme/default.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  pname = "gtkme";
+  version = "1.5.3";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-NIUgnbfcHjbPfsH3CF2Bywo8owrdsi1wqDoMxOa+2U4=";
+  };
+
+  nativeBuildInputs = [ pkg-config pkgs.gobject-introspection ];
+  buildInputs = [ pkgs.python27Packages.pygobject3 pkgs.gtk3 ];
```
Comment (veprbl) [inline pkgs/development/python2-modules/gtkme/default.nix:19]:
```suggestion
  buildInputs = [ pygobject3 pkgs.gtk3 ];
```

File: pkgs/development/python2-modules/gtkme/default.nix [hunk 4]
```diff
@@ -0,0 +1,29 @@
+{ pkgs
+, lib
+, pkg-config
+, buildPythonPackage
+, buildPackages
```
Comment (veprbl) [inline pkgs/development/python2-modules/gtkme/default.nix:5]:
```suggestion
```

File: pkgs/development/python2-modules/gtkme/default.nix (hunk 3 above)
Comment (veprbl) [inline pkgs/development/python2-modules/gtkme/default.nix:19]: Actually
```suggestion
  nativeBuildInputs = [ pkg-config gobject-introspection ];
  buildInputs = [ pygobject3 gtk3 ];
```

File: pkgs/development/python2-modules/gtkme/default.nix [hunk 5]
```diff
@@ -0,0 +1,29 @@
+{ pkgs
+, lib
```
Comment (veprbl) [inline pkgs/development/python2-modules/gtkme/default.nix:2]:
```suggestion
{ lib
```

File: pkgs/development/python2-modules/gtkme/default.nix [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    sha256 = "sha256-NIUgnbfcHjbPfsH3CF2Bywo8owrdsi1wqDoMxOa+2U4=";
+  };
+
+  nativeBuildInputs = [ pkg-config pkgs.gobject-introspection ];
+  buildInputs = [ pkgs.python27Packages.pygobject3 pkgs.gtk3 ];
+
+  meta = with lib; {
+    description = "Manages an Application with Gtk windows, forms, lists and other complex items easily.";
+    homepage = "https://gitlab.com/doctormo/gtkme";
+    license = licenses.gpl3;
```
Comment (veprbl) [inline pkgs/development/python2-modules/gtkme/default.nix:24]:
```suggestion
    license = licenses.gpl3Plus;
```

File: pkgs/development/python2-modules/gtkme/default.nix [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-NIUgnbfcHjbPfsH3CF2Bywo8owrdsi1wqDoMxOa+2U4=";
+  };
+
+  nativeBuildInputs = [ pkg-config pkgs.gobject-introspection ];
+  buildInputs = [ pkgs.python27Packages.pygobject3 pkgs.gtk3 ];
+
+  meta = with lib; {
+    description = "Manages an Application with Gtk windows, forms, lists and other complex items easily.";
```
Comment (veprbl) [inline pkgs/development/python2-modules/gtkme/default.nix:22]:
```suggestion
    description = "Manages an Application with Gtk windows, forms, lists and other complex items easily";
```

File: pkgs/development/python2-modules/gtkme/default.nix (hunk 3 above)
Comment (revol-xut) [reply pkgs/development/python2-modules/gtkme/default.nix:19]: you mean I should add them as input to the derivation.

Comment (veprbl) [reply pkgs/development/python2-modules/gtkme/default.nix:19]: Right. This is the preferred way, as it allows to do individual overrides

Comment (revol-xut) [reply pkgs/development/python2-modules/gtkme/default.nix:19]: ok thx for the advice

**Comment by veprbl** [comment]
So how does this work?

```diff
diff --git a/pkgs/development/python2-modules/gtkme/default.nix b/pkgs/development/python2-modules/gtkme/default.nix
index 71f88a74e46..1099e2c2d4b 100644
--- a/pkgs/development/python2-modules/gtkme/default.nix
+++ b/pkgs/development/python2-modules/gtkme/default.nix
@@ -19,6 +19,8 @@ buildPythonPackage rec {
   nativeBuildInputs = [ pkg-config gobject-introspection ];
   buildInputs = [ pygobject3 gtk3 ];
 
+  pythonImportsCheck = [ "gtkme" ];
+
   meta = with lib; {
     description = "Manages an Application with Gtk windows, forms, lists and other complex items easily";
     homepage = "https://gitlab.com/doctormo/gtkme";
```
gives
```
Executing pythonImportsCheckPhase
Check whether the following modules can be imported: gtkme
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "<string>", line 1, in <lambda>
  File "/nix/store/0qjdc4ilywc7rkvdc59qjr6llilhkkv3-python-2.7.18/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
  File "/nix/store/f1ns2n2wk92vndq5hxg6jkk70crnk95z-python2.7-gtkme-1.5.3/lib/python2.7/site-packages/gtkme/__init__.py", line 35, in <module>
    gi.require_version('Gtk', '3.0')
  File "/nix/store/99zbidms5hmzlp2hlzrlnwzi3kf9rjh1-python2.7-pygobject-3.36.1/lib/python2.7/site-packages/gi/__init__.py", line 129, in require_version
    raise ValueError('Namespace %s not available' % namespace)
ValueError: Namespace Gtk not available
```

**Comment by revol-xut** [comment]
@veprbl I fixed that it was missing `gtk3` as a `nativeBuildInput`. Probably couldn't find some headers gtk-devel headers.

File: pkgs/development/python2-modules/gtkme/default.nix [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+  version = "1.5.3";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-NIUgnbfcHjbPfsH3CF2Bywo8owrdsi1wqDoMxOa+2U4=";
+  };
+
+  nativeBuildInputs = [ pkg-config gobject-introspection ];
+  buildInputs = [ pygobject3 gtk3 ];
+
```
Comment (veprbl) [inline pkgs/development/python2-modules/gtkme/default.nix:21]:
```suggestion

  pythonImportsCheck = [ "gtkme" ];

```

**Comment by veprbl** [comment]
> @veprbl I fixed that it was missing `gtk3` as a `nativeBuildInput`. Probably couldn't find some gtk-devel headers.

I can confirm, that works. You did not implement the change in the last force push. Also I wonder if it would have to go to `checkInputs` or `propagatedBuildInputs` instead?

**Comment by revol-xut** [comment]
I would make sense to have `gtk3` as a `propagatedBuildInput` because the shared library is also needed downstream.

**Comment by veprbl** [comment]
@GrahamcOfBorg build python27Packages.gtkme

**Review by veprbl — APPROVED** [review]

File: pkgs/development/python2-modules/gtkme/default.nix [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+  version = "1.5.3";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-NIUgnbfcHjbPfsH3CF2Bywo8owrdsi1wqDoMxOa+2U4=";
+  };
+
+  nativeBuildInputs = [ pkg-config gobject-introspection gtk3 ];
+  buildInputs = [ pygobject3 glib ];
+  propagatedBuildInputs = [ gtk3 ];
```
Comment (dotlambda) [inline pkgs/development/python2-modules/gtkme/default.nix:22]:
```suggestion
  buildInputs = [ glib gtk3 ];
  propagatedBuildInputs = [ pygobject3 ];
```

**Pull request merged**